File: Algorithm/Programers_바탕화면 정리.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<string> wallpaper) {
	vector<int> answer;
	int minX = 50;
	int maxX = 0;
	int minY = 50;
	int maxY = 0;
	for (int i = 0; i < wallpaper.size(); i++) {
		for (int j = 0; j < wallpaper[i].size(); j++) {
			if (wallpaper[i][j] != '#') continue;
			if (i < minX) minX = i;
			if (i > maxX) maxX = i;
			if (j < minY) minY = j;
			if (j > maxY) maxY = j;

		}
	}
	answer.push_back(minX);
	answer.push_back(minY);
	answer.push_back(maxX + 1);
	answer.push_back(maxY + 1);
	return answer;
}
//https://school.programmers.co.kr/learn/courses/30/lessons/161990?language=cpp

File: Algorithm/Programers_바탕화면_정리 .java
import java.util.*;
class Solution {
    public int[] solution(String[] wallpaper) {
        int MAX_X=Integer.MIN_VALUE,MIN_X=Integer.MAX_VALUE,MAX_Y=Integer.MIN_VALUE,MIN_Y=Integer.MAX_VALUE;
        for(int y=0;y< wallpaper.length;y++) {
            for (int x = 0; x < wallpaper[y].length(); x++) {
                if (wallpaper[y].charAt(x) != '#')
                    continue;

                MAX_X = Math.max(MAX_X,x);
                MIN_X = Math.min(MIN_X,x);
                MAX_Y = Math.max(MAX_Y,y);
                MIN_Y = Math.min(MIN_Y,y);

            }
        }
        int[] answer = {MIN_Y, MIN_X, MAX_Y + 1, MAX_X + 1};
        return answer;
    }
}
class main{
    public static void main(String[] args) {
        Solution sol = new Solution();
        String wallpaper[] = {".#...", "..#..", "...#."};
        System.out.println("sol = " + sol.solution(wallpaper));
    }
}
//https://school.programmers.co.kr/learn/courses/30/lessons/161990